python/cudf/cudf/core/subword_tokenizer.py

```diff
@@ -0,0 +1,279 @@
+                [ 101, 1436, 1520,  102,    0,    0,    0,    0]],
+                device='cuda:0',
+               dtype=torch.int32)
+        >>>tokenizer_output['attention_mask']
+        tensor([[1, 1, 1, 1, 1, 0, 0, 0],
+                [1, 1, 1, 1, 0, 0, 0, 0]],
+                device='cuda:0', dtype=torch.int32)
+
+        >>>tokenizer_output['metadata']
+        tensor([[0, 1, 3],
+                  [1, 1, 2]], device='cuda:0', dtype=torch.int32)
+       """
+
+        if return_token_type_ids:
+            # raise not currently supported
+            # Can also return zeros
+            error_msg = "Returning token_type_ids is currently supported"
+            raise NotImplementedError(error_msg)
+
+        if truncation in (False, "do_not_truncate"):
+            truncation = False
+            warning_msg = (
+                "When truncation is not True, the behaviour currently differs "
+                "from HuggingFace as cudf always returns overflowing tokens"
+            )
+            warn(warning_msg)
+            if add_special_tokens:
+                error_msg = (
+                    f"Adding special tokens is not supported"
+                    f"with truncation = {truncation}. "
+                )
+                recommendation = (
+                    "Custom Cupy kernel can potentially"
+                    "be used to add it. For reference "
+                    "see: _bert_add_special_tokens "
+                )
+                raise NotImplementedError(error_msg + recommendation)
+
+        if padding != "max_length":
+            error_msg = (
+                "Currently padding to the provided max_length is supported"
+            )
+            raise NotImplementedError(error_msg)
+
+        if max_length <= stride:
+            error_msg = "Stride should be less than max_length"
+            raise ValueError(error_msg)
+
+        if return_tensors not in ["cp", "pt", "tf"]:
+            error_msg = (
+                "Only cupy(cp), pytorch(pt) and tensorflow(tf)"
+                "tensors are supported"
+            )
+            raise NotImplementedError(error_msg)
+
+        stride = max_length - stride
+        # behaviour varies from subword_tokenize but maps with huggingface
+
+        input_ids, attention_mask, metadata = cpp_subword_tokenize(
+            text._column,
+            self.vocab_file,
+            max_sequence_length=max_length,
+            stride=stride,
+            do_lower=self.do_lower_case,
+            do_truncate=truncation,
+            max_rows_tensor=max_num_rows,
+        )
+
+        tokenizer_output = dict()
+        tokenizer_output["input_ids"] = cp.asarray(input_ids).reshape(
+            -1, max_length
+        )
+        tokenizer_output["attention_mask"] = cp.asarray(
+            attention_mask
+        ).reshape(-1, max_length)
+        tokenizer_output["metadata"] = cp.asarray(metadata).reshape(-1, 3)
```

Comment:
Should just initialize the dictionary:
```suggestion
        tokenizer_output = {
            "input_ids": cp.asarray(input_ids).reshape(
                -1, max_length
            )
            "attention_mask": cp.asarray(
                attention_mask
            ).reshape(-1, max_length)
            "metadata": cp.asarray(metadata).reshape(-1, 3)
        }
```

Reply:
Resolved with commit : https://github.com/rapidsai/cudf/commit/0018779bde5984931e114a3cd412ce869706a7b3